src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -0,0 +1,229 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Device.Gpio;
+using System.Device.Spi;
+
+namespace Iot.Device.Spi
+{
+    public class SoftwareSpi : SpiDevice
```

Comment:
We currently have [SoftPwm](https://github.com/dotnet/iot/tree/master/src/devices/SoftPwm).  This needs to be renamed `SoftSpi` or `SoftPwm` needs to be renamed to `SoftwarePwm` to be consistent.

Reply:
actually SoftPwm is since couple of days ago SoftwarePwmChannel, we just didn't rename the folder

Reply:
I like this being SoftwareSpi and the other one being SoftwarePwmChannel, I agree we should just rename the folder of SoftPwm to match the binding.

Reply:
>  I agree we should just rename the folder of SoftPwm to match the binding.

Let's hold off on the SoftPwm rename.  I wanted to test some and I'm assuming it will be moved to core API and hidden with its own Create(... pin params ...) based on previous talks.

Reply:
> I like this being SoftwareSpi...

Since this will also eventually be hidden and placed in core API with its own Create(... pin params ...), what would be the harm of naming `SoftwareSpiDevice` as it will implement SpiDevice?  Just curious.

Reply:
> I wanted to test some and I'm assuming it will be moved to core API and hidden with its own Create(... pin params ...) based on previous talks.

This is in the future plans but not for 3.0, so What I was referring above was simply renaming the folder under devices. Same goes for SoftwareSpi, we don't want to move that under System.Device.Gpio library just yet but do think that we will do so in the future.

Reply:
@joperezr gotcha.  i'll try to rename the SoftPwm > SoftwarePwm later in the week then.